### PR TITLE
Scheduler predicate test explicitly assumes that cluster add-on pods …

### LINKED ISSUE #19895: e2e failure: SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run [Conformance]
e2e failure: SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run [Conformance]
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:355 Expected : 21 to equal : 22
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce/10440/

@saad-ali @hongchaodeng Recent issues/PRs mentioning the same test 
- #19487 
- #19495

### PR DESCRIPTION
**Pull request by dchen1107**
…stay stable, but never check it before each test run. Adding the check for each test case.

Fixed #19895

### CODE REVIEW

File: test/e2e/scheduler_predicates.go [hunk 1]
```diff
@@ -195,6 +196,13 @@ var _ = Describe("SchedulerPredicates [Serial]", func() {
 		c = framework.Client
 		ns = framework.Namespace.Name
 		nodeList = ListSchedulableNodesOrDie(c)
+
+		systemPods, err := c.Pods(api.NamespaceSystem).List(api.ListOptions{})
```
Comment (ixdy) [inline test/e2e/scheduler_predicates.go]: Maybe add a short comment explaining this?

**Comment by dchen1107** [comment]
@ixdy Adding comments per your suggestion.

**Comment by ixdy** [comment]
LGTM

**Comment by dchen1107** [comment]
Merged this one manually for test flakiness fix.

**Comment by dchen1107** [comment]
Filed #20910 (e2e flaky: should be able to change the type and ports of a service) which has nothing to do with this change. 

ok to test

**Pull request merged**